client.go

```diff
@@ -0,0 +1,120 @@
+	"fmt"
+	"log"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/pkg/errors"
+	"perun.network/go-perun/apps/payment"
+	"perun.network/go-perun/backend/ethereum/channel"
+	"perun.network/go-perun/backend/ethereum/wallet"
+	"perun.network/go-perun/client"
+	"perun.network/go-perun/peer/net"
+	perunwallet "perun.network/go-perun/wallet"
+)
+
+type Client struct {
+	cfg *Config
+
+	client *client.Client
+
+	w       *wallet.Wallet
+	onChain perunwallet.Account
+
+	dialer   *net.Dialer
+	listener *net.Listener
+}
+
+func NewClient(cfg *Config) (*Client, error) {
+	//acc := unlockAccount(cfg)
+	//var w wallet.Wallet
+	w, acc, err := importAccount(cfg.KeyStorePath, "0x6aeeb7f09e757baa9d3935a042c3d0d46a2eda19e9b676283dce4eaf32e29dc9")
+	if err != nil {
+		return nil, errors.WithMessage(err, "importing account")
+	}
+	endpoint := fmt.Sprintf("%s:%d", cfg.IP, cfg.Port)
+	listener, err := net.NewTCPListener(endpoint)
+	if err != nil {
+		return nil, errors.WithMessagef(err, "listening on %s", endpoint)
+	}
+	dialer := net.NewTCPDialer(time.Second * 15)
+	node, err := ethclient.Dial(cfg.ETHNodeURL)
+	if err != nil {
+		return nil, errors.WithMessage(err, "connecting to ethereum node")
+	}
+	cb := channel.NewContractBackend(node, w.Ks, &acc.Account)
+	adjudicator := channel.NewAdjudicator(cb, adjudicatorAdr, acc.Account.Address)
+	funder := channel.NewETHFunder(cb, assetAddr)
+
+	client := client.New(acc, dialer, funder, adjudicator, w)
+	return &Client{cfg: cfg, client: client, w: w, onChain: acc, dialer: dialer, listener: listener}, nil
+}
+
+func unlockAccount(cfg *Config) wallet.Account {
+	return wallet.Account{}
+}
+
+func importAccount(walletPath, secret string) (*wallet.Wallet, *wallet.Account, error) {
+	ks := keystore.NewKeyStore(walletPath, 2, 1)
+	sk, err := crypto.HexToECDSA(secret[2:])
+	if err != nil {
+		return nil, nil, errors.WithMessage(err, "decoding secret key")
+	}
+	var ethAcc accounts.Account
+	addr := crypto.PubkeyToAddress(sk.PublicKey)
+	if ethAcc, err = ks.Find(accounts.Account{Address: addr}); err != nil {
+		ethAcc, err = ks.ImportECDSA(sk, "")
+		if err != nil && errors.Cause(err).Error() != "account already exists" {
+			return nil, nil, errors.WithMessage(err, "importing secret key")
+		}
+	}
+
+	w, err := wallet.NewWallet(ks, "")
+	if err != nil {
+		return nil, nil, errors.WithMessage(err, "creating wallet")
+	}
+
+	wAcc := wallet.NewAccountFromEth(w, &ethAcc)
+	acc, err := w.Unlock(wAcc.Address())
+	return w, acc.(*wallet.Account), err
+}
+
+func (c *Client) AddPeer(perunID *Address, ip string, port int) {
+	c.dialer.Register((*wallet.Address)(&perunID.Address), fmt.Sprintf("%s:%d", ip, port))
+}
+
+func (c *Client) ProposeChannel(ctx *Context, perunID *Address, initialBals *Allocation, challengeDuration int64) (*PaymentChannel, error) {
```

Comment:
```suggestion
func (c *Client) ProposeChannel(ctx *Context, peerPerunID *Address, challengeDuration int64, initialBals BigInts) (*PaymentChannel, error) {
```
I took me a while to understand what `perunID` is. I'd also exchange the duration and bals so we start with the parameters and then specify the initial state.

Reply:
and, as mentioned before, `Allocation` should at this point not be part of the mobile api and we just use `BigInts`.